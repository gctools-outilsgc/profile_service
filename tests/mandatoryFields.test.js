const mutations = require("../src/resolvers/Mutations");
const { getPrismaTestInstance } = require("./init/prismaTestInstance");

const parent = {};
const ctx = {
    prisma: getPrismaTestInstance()
};


test("create a basic profile with mandatory fields",  async() => {
    const args = {
            gcID: "09ujilkjlkiid",
            name: "Awesome User",
            email: "awesome.user@somewhere.com"
    };

    const info = "{ gcID, name, email }";

    expect(
        await mutations.createProfile(parent, args, ctx, info),
    ).toMatchSnapshot();

    await getPrismaTestInstance().mutation.deleteProfile({where:{gcID:"09ujilkjlkiid"}});
});

test("fail profile creation due to missing gcID", async() => {
    const args = {
        name: "Awesome User",
        email: "awesome.user@somewhere.com"
    };
     
    const info = "{ gcID, name, email }";
    
    await expect(
        mutations.createProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();

});

test("fail profile creation due to missing name", async() => {

    const args = {
            gcID: "09ujilkjlkiid",
            email: "awesome.user@somewhere.com"
        };
               
        const info = "{ gcID, name, email }";

    await expect(
        mutations.createProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();
});

test("fail profile creation due to missing email", async() => {

        const args = {
            gcID: "09ujilkjlkiid",
            name: "Awesome User",
        };
        
        const info = "{ gcID, name, email }";
        
    await expect(
        mutations.createProfile(parent, args, ctx, info)
    ).rejects.toThrowErrorMatchingSnapshot();

});