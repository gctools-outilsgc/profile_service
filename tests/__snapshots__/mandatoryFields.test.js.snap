// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a basic profile with mandatory fields 1`] = `
Object {
  "email": "awesome.user@somewhere.com",
  "gcID": "09ujilkjlkiid",
  "name": "Awesome User",
}
`;

exports[`fail profile creation due to missing email 1`] = `"Variable \\"$_v0_data\\" got invalid value { gcID: \\"09ujilkjlkiid\\", name: \\"Awesome User\\", email: undefined, mobilePhone: undefined, officePhone: undefined, titleEn: undefined, titleFr: undefined }; Field value.email of required type String! was not provided."`;

exports[`fail profile creation due to missing gcID 1`] = `"Variable \\"$_v0_data\\" got invalid value { gcID: undefined, name: \\"Awesome User\\", email: \\"awesome.user@somewhere.com\\", mobilePhone: undefined, officePhone: undefined, titleEn: undefined, titleFr: undefined }; Field value.gcID of required type String! was not provided."`;

exports[`fail profile creation due to missing name 1`] = `"Variable \\"$_v0_data\\" got invalid value { gcID: \\"09ujilkjlkiid\\", name: undefined, email: \\"awesome.user@somewhere.com\\", mobilePhone: undefined, officePhone: undefined, titleEn: undefined, titleFr: undefined }; Field value.name of required type String! was not provided."`;
